python/lsst/ap/pipe/createApFakes.py

```diff
@@ -0,0 +1,300 @@
+        default="isVisitSource"
+    )
+    templateSourceFlagCol = pexConfig.Field(
+        doc="Name of the column flagging objects for insertion into the "
+            "template image.",
+        dtype=str,
+        default="isTemplateSource"
+    )
+
+
+class CreateRandomApFakesTask(PipelineTask):
+    """Create and store a set of spatially uniform fakes for use in AP
+    processing.
+    """
+
+    _DefaultName = "createApFakes"
+    ConfigClass = CreateRandomApFakesConfig
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        inputs["tractId"] = butlerQC.quantum.dataId["tract"]
+
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, tractId, skyMap):
+        """Create a set of uniform random points that covers a tract.
+
+        Parameters
+        ----------
+        tractId : `int`
+            Tract id to produce randoms over.
+        skyMap : `lsst.skymap.SkyMap`
+            Skymap to produce randoms over.
+
+        Returns
+        -------
+        randoms : `lsst.pipe.tasks.PaquertTable`
+            Catalog of random points covering the given tract. Follows the
+            format expected in `lsst.pipe.tasks.InsertFakes`.
+        """
+        np.random.seed(self.config.randomSeed)
+        tractBoundingCircle = \
+            skyMap.generateTract(tractId).getInnerSkyPolygon().getBoundingCircle()
+        tractArea = tractBoundingCircle.getArea() * (180 / np.pi) ** 2
+        nFakes = int(self.config.fakeDensity * tractArea)
+
+        self.log.info(
+            f"Creating {nFakes} over tractId={tractId} with BBox area: "
+            "{tractArea} deg^2")
+
+        randData = {
+            **self.createRandomPositions(nFakes, tractBoundingCircle),
+            **self.createVisitCoaddSubdivision(nFakes),
+            **self.createRandomMagnitudes(nFakes),
+            self.config.diskHLR: np.ones(nFakes, dtype="float"),
+            self.config.bulgeHLR: np.ones(nFakes, dtype="float"),
+            self.config.nDisk: np.ones(nFakes, dtype="float"),
+            self.config.nBulge: np.ones(nFakes, dtype="float"),
+            self.config.aDisk: np.ones(nFakes, dtype="float"),
+            self.config.aBulge: np.ones(nFakes, dtype="float"),
+            self.config.bDisk: np.ones(nFakes, dtype="float"),
+            self.config.bBulge: np.ones(nFakes, dtype="float"),
+            self.config.paDisk: np.ones(nFakes, dtype="float"),
+            self.config.paBulge: np.ones(nFakes, dtype="float")}
+
+        return Struct(
+            fakeCat=ParquetTable(dataFrame=pd.DataFrame(data=randData)))
+
+    def createRandomPositions(self, nFakes, boundingCircle):
+        """Create a set of spatially uniform randoms over the tract bounding
+        circle.
+
+        Parameters
+        ----------
+        nFakes : `int`
+            Number of fakes to create.
+        boundingCicle : `lsst.sphgeom.BoundingCircle`
```

Comment:
As far as I can see, all the random generation and transformation support up to a full sphere (up to Pi opening angle). Be explicit in the docstring about the maximum size of the allowed `BoundingCircle`.

Reply:
A larger than Pi value for a bounding circle doesn't make sense as a bounding circle of Pi is already encompasses the full sphere.